subsys/net/lib/openthread/platform/radio.c

```diff
@@ -309,9 +378,33 @@ int8_t otPlatRadioGetRssi(otInstance *aInstance)
 
 otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
 {
+	otRadioCaps caps = OT_RADIO_CAPS_NONE;
+
+	enum ieee802154_hw_caps radio_caps;
 	ARG_UNUSED(aInstance);
+	__ASSERT(radio_dev,
+	    "platformRadioInit needs to be called prior to otPlatRadioGetCaps");
+	__ASSERT(radio_api,
+	    "platformRadioInit needs to be called prior to otPlatRadioGetCaps");
+
+	radio_caps = radio_api->get_capabilities(radio_dev);
+	if (radio_caps & IEEE802154_HW_TX_RX_ACK) {
+		/*
+		 * In current state of nrf radio driver ack timeout is not
```

Comment:
We should not mention nRF radio driver here, it's a generic layer. IMO we should only focus on `OT_RADIO_CAPS_ENERGY_SCAN` in this PR.